### PR TITLE
Allow pydantic v2 using transitional v1 support

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Allow pydantic v2 using transitional v1 support.

### Types of change

Enhancement?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
@explosion-bot please test_slow

**Comment by adrianeboyd** [comment]
@explosion-bot please test_slow

**Comment by adrianeboyd** [comment]
wheelwright build (with some hacks to use pydantic v1 for testing on aarch64) is fine:

https://dev.azure.com/explosion-ai/Public/_build/results?buildId=26594&view=results

File: spacy/pipeline/_edit_tree_internals/schemas.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 from typing import Any, Dict, List, Union
 
-from pydantic import BaseModel, Field, ValidationError
-from pydantic.types import StrictBool, StrictInt, StrictStr
+try:
+    from pydantic.v1 import BaseModel, Field, ValidationError
+    from pydantic.v1.types import StrictBool, StrictInt, StrictStr
+except ImportError:
+    from pydantic import BaseModel, Field, ValidationError  # type: ignore
+    from pydantic.types import StrictBool, StrictInt, StrictStr  # type: ignore
```
Comment (rmitsch) [inline spacy/pipeline/_edit_tree_internals/schemas.py:9]: `from pydantic.v1` will error if `pydantic` 1.x is installed, but work with `pydantic 2.x`, right?

Comment (adrianeboyd) [reply spacy/pipeline/_edit_tree_internals/schemas.py:9]: Yes

**Review by rmitsch — APPROVED** [review]

**Comment by meerfrau** [comment]
Are there specific instructions for using this?

I'm still getting `ImportError: cannot import name 'ModelMetaclass' from 'pydantic.main' (/usr/lib/python3.10/site-packages/pydantic/main.py)` with Pydantic 2 (both 2.0.3 or 2.3.0) w/ Py3.10

**Comment by adrianeboyd** [comment]
This is the error you would get if you have pydantic v2 installed with spacy v3.6.0 or earlier. Can you check with `python -m pip check` if there are any version conflicts for the installed packages?

**Comment by meerfrau** [comment]
I've switched back to pydantic v1

**Comment by maspotts** [comment]
I also am getting:

`ImportError: cannot import name 'ModelMetaclass' from 'pydantic.main' (/Users/mike/.pyenv/versions/3.10.7/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pydantic/main.py)
`

with pydantic 2.6.1 and spacy 3.7.4.  what am I doing wrong?

**Pull request merged**